ql/src/java/org/apache/hadoop/hive/ql/DriverUtils.java

```diff
@@ -95,7 +95,7 @@ public static SessionState setUpSessionState(HiveConf conf, String user, boolean
     if (sessionState == null) {
       // Note: we assume that workers run on the same threads repeatedly, so we can set up
       //       the session here and it will be reused without explicitly storing in the worker.
-      sessionState = new SessionState(conf, user);
+      sessionState = new SessionState(conf, user, true);
```

Comment:
Are we targeting specific queries like auto-gather background stats threads & compaction ? why are we not providing a config to toggle sync/async delete ?

Reply:
background threads do not need async delete. Many compaction tests specifically have sync assumptions. I dont see any benefit in moving background operations to async cleanup model.